src/module/actor/creature/sheet.ts

```diff
@@ -475,6 +475,25 @@ export abstract class CreatureSheetPF2e<TActor extends CreaturePF2e> extends Act
                 await effect.increase();
             }
         });
+
+        // Change whether an effect is secret to players or not
+        for (const effects_sections of htmlQueryAll(html, ".effects-list") ?? []) {
+            for (const element of htmlQueryAll(effects_sections, "[data-action=effect-toggle-secret]") ?? []) {
```

Comment:
Replace with a single loop for `htmlQueryAll(html, ".effects-list [data-action=effect-toggle-secret]")`. You should be able to do more or less the same queries you did with Jquery.

Reply:
I appreciate the trust you put in my knowledge of JQuery. :sweat_smile: